addons/sms/wizard/sms_composer_views.xml

```diff
@@ -48,8 +48,8 @@
                             <field name="recipient_single_number_itf" class="oe_inline" nolabel="1" options="{'onchange_on_keydown': True}" placeholder="e.g. +1 415 555 0100"/>
                         </div>
                                                 
-                        <field name="body" widget="sms_widget"/>
-                        <field name="mass_keep_log" invisible="1"/>
+                        <field name="body" widget="sms_widget" attrs="{'invisible': [('comment_single_recipient', '=', True), ('recipient_single_valid', '=', True)]}"/>
+                        <field name="body" widget="sms_widget" attrs="{'invisible': ['|', ('comment_single_recipient', '=', False), '&amp;', ('comment_single_recipient', '=', True), ('recipient_single_valid', '=', False)]}" default_focus="1"/>
```

Comment:
"not(A and B)" should be "not A or not B", or am I mistaken ?

Reply:
We have two condition for the OR part here. Not sure what you're trying to say.

Reply:
Second body is basically the inverse of first body. So I think domain of second one could be simplified.